src/request-pipeline/context.ts

```diff
@@ -284,13 +284,14 @@ export default class RequestPipelineContext {
     }
 
     redirect (url: string) {
-        if ('setHeader' in this.res) {
-            this.res.statusCode = 302;
-            this.res.setHeader('location', url);
-            this.res.end();
-        }
-        else
-            throw new Error('???');
+        if (this.isWebSocket)
+            throw new Error('The RequestPipelineContext.prototype.redirect function cannot be used with a WebSocket request');
```

Comment:
Cut the error message to `The function cannot be used with a WebSocket request`.
Because the class name and the function can be changed. And you can detect the problem place using a call stack.